src/coreclr/tools/Common/TypeSystem/IL/Stubs/CalliMarshallingMethodThunk.cs

```diff
@@ -21,11 +21,13 @@ public partial class CalliMarshallingMethodThunk : ILStubMethod
         private MethodSignature _signature;
 
         public CalliMarshallingMethodThunk(MethodSignature targetSignature, TypeDesc owningType,
-                InteropStateManager interopStateManager)
+                InteropStateManager interopStateManager,
+                ModuleDesc moduleContext)
```

Comment:
It would be better to have a `bool marshallingSupported` flag here.

These thunks can be shared for the same signature; having a module here limits the sharing (we can end up with multiple thunks that only differ by their module, but the only thing that affects the code in these is whether the module has an attribute).

This will also need a change in CalliMarshallingMethodThunk.Mangling.cs (the thunks need a mangled name in the object file and we're now at risk of collisions - the prefix should incorporate the boolean flag) and a change in CalliMarshallingMethodThunk.Sorting.cs (this is used to get a stable order of outputs after we finished multithreaded compilation - right now signatures that differ only by marshalling state would sort equal).